src/fraction.ts

```diff
@@ -0,0 +1,88 @@
+import BN from "bn.js";
+
+export class Fraction {
+  private numerator: BN;
+  private denominator: BN;
```

Comment:
Maybe these can be `readonly` instead of private?

Reply:
The problem is the `reduce` method, where I reassign the numerator/denominator. Do you think I should return a new Fraction there as well?

Reply:
Hmmm, no I think you are right.